widgetastic_manageiq/__init__.py

```diff
@@ -1855,29 +1855,290 @@ def is_displayed(self):
 
-    def clear_search(self):
+    clear_button = Text(".//*[@id='searchbox']//div[contains(@class, 'clear') "
+                        "and not(contains(@style, 'display: none'))]/div/button")
+    filter_clear_button = Text('//a[contains(@href, "adv_search_clear")]')
+    advanced_search_button = Button(title='Advanced Search')
+
+    advanced_search_form = View.nested(AdvancedSearchView)
+
+# ================================= Simple Search ====================================
+
+    @property
+    def has_quick_search_box(self):
+        return self.search_input.is_displayed
+
+    def clear_simple_search(self):
+        """ Clear simple search field """
         if not self.is_empty:
-            self.clear_btn.click()
-            self.search_btn.click()
+            self.clear_button.click()
+            self.search_button.click()
 
-    def search(self, text):
-        self.search_text.fill(text)
-        self.search_btn.click()
+    def simple_search(self, text):
+        """ Search text using simple search """
+        self.search_input.fill(text)
+        self.search_button.click()
 
     @property
     @logged(log_result=True)
     def is_empty(self):
-        return not bool(self.search_text.value)
+        """ Checks if simple search field is emply """
+        return not bool(self.search_input.value)
+
+
+# ================================= Advanced Search ===============================
+    @property
+    def is_advanced_search_opened(self):
+        """Checks whether the advanced search box is currently opened"""
+        return self.advanced_search_form.is_displayed
+
+    def reset_filter(self):
+        """Clears the filter expression
+
+            Returns: result of clicking reset when enabled(True),
+                false when reset is button is disabled
+        """
+        view = self.advanced_search_form
+        if not view.reset_filter_button.disabled:
+            view.reset_filter_button.click()
+            reset_result = True
+        else:
+            reset_result = False
+        view.close_button.click()
+        return reset_result
+
+    def apply_filter(self):
+        """ Applies an existing filter
+
+            Returns: Apply button state, True - if active and clicked,
+                False - disabled, or not visible
+        """
+        try:
+            self.advanced_search_form.apply_filter_button.click()
+            return True
+        except NoSuchElementException:
+            return False
+
+    def delete_filter(self, cancel=False):
+        """If possible, deletes the currently loaded filter
+
+            Returns: Delet button state, True - if active and clicked,
+                False - disabled, or not visible
+        """
+        try:
+            self.advanced_search_form.delete_filter_button.click(handle_alert=not cancel)
+            return True
+        except NoSuchElementException:
+            return False
```

Comment:
**COMMENT (no change required in this PR)**
`apply_filter` and `delete_filter` shouldn't eat up the exception. They should raise it further or raise a more specific one, if needed. I can imagine `delete_filter(quite=True)` for cleanup purposes but that's about it.